falcon/asgi/app.py

```diff
@@ -521,11 +477,17 @@ async def __call__(self, scope, receive, send):  # noqa: C901
             })
 
             await send(_EVT_RESP_EOF)
-            self._schedule_callbacks(resp)
+
+            # PERF(vytas): Check resp._registered_callbacks directly to shave
+            #   off a function call since this is a hot/critical code path.
+            if resp._registered_callbacks:
+                self._schedule_callbacks(resp)
```

Comment:
can't we just pass the cached `_registered_callbacks` to the function? it does not otherwise uses resf if I'm not mistaken

Reply:
I don't know if it is safe, I want to wrap this up somehow.

Like what happens if the `resp.stream` generator chooses to add schedule more callbacks along the way?
It's not trivial when to acquire the value to cache.